test/parallel/test-process-exit-code-validation.js

```diff
@@ -141,3 +138,10 @@ if (process.argv[2] === undefined) {
     process.exit(args[index].code);
   }
 }
+
+const origExitCode = Object.getOwnPropertyDescriptor(process, 'exitCode');
+try {
+  delete process.exitCode;
```

Comment:
I'm not sure how that would be possible. If it's deleted, there's nothing there.

Reply:
Do note that `process.exitCode = kGenericUserError; ` will still work after it's been deleted.

Reply:
If `process.exitCode = kGenericUserError;` worked, the error code after terminating the node should be `kGenericUserError`, not 0. Last time I checked, it was 0... right?

Reply:
I mean it should not be 0 when calling the exit status. `echo $?`

Reply:
ahh, i see what you mean. since node doesn't *read* from `process.exitCode` anymore, once it's deleted, it's impossible to set the exitcode.

I don't think this is an avoidable problem unless node always reads `process.exitCode` (instead of using the `fields` private storage).

Reply:
Is that a reasonable solution? in other words, when it's time to read the exit code, if process.exitCode is a data property, use its value; otherwise use the privately stored value?

Reply:
I think it's possible, but if not considering a breaking change, there are more things to think about.

- Integer coercion policy for a data property which isn't allowed. (e.g., 'any')
- Validating it and throwing if it cannot be written as the exit status.

Reply:
What about a data property, whose value is an integer number >= 0, else the private value?

Reply:
I don't catch your intent very well. AFAICT, negative integer values are also commonly used, and they are not the range of the exit code on Windows.

Reply:
it can be restricted to any domain you like :-) just integer numbers then?

Reply:
ok, so I’ll move forward with that schema, thanks.

Reply:
hmm - seems like the exitCode is read in C++, so i'm not sure how it can read the JS-set value once it's overridden.

@daeyeon are you comfortable with this proceeding with the understanding that if a user mocks process.exitCode, it's up to them to restore it? (just like other globals)

If not, what about https://github.com/nodejs/node/pull/49579#issuecomment-2232295710 ?

Reply:
> @daeyeon are you comfortable with this proceeding with the understanding that if a user mocks process.exitCode, it's up to them to restore it? (just like other globals)

The core's behavior related to exitCode will be breaking during the time gap between mocking and restoring. It's probably not enough to rely on the restoring after finishing the mocking.

> If not, what about https://github.com/nodejs/node/pull/49579#issuecomment-2232295710 ?

Looks like a good idea. In addition to moving the getters/setters to the process prototype, I think that the core code should use the exitCode of the process prototype, rather than the process itself. This would address concerns about breaking and can provide guidance on how users can monitor it.